proc/process.c

```diff
@@ -1036,12 +1039,14 @@ static void process_vforkThread(void *arg)
 	pmap_switch(current->process->pmapp);
 
 	hal_spinlockSet(&spawn->sl, &sc);
-	while (spawn->state < FORKING)
+	while (spawn->state < FORKING) {
 		proc_threadWait(&spawn->wq, &spawn->sl, 0, &sc);
+	}
 	hal_spinlockClear(&spawn->sl, &sc);
 
 	/* Copy parent kernel stack */
-	if ((current->parentkstack = (void *)vm_kmalloc(parent->kstacksz)) == NULL) {
+	current->parentkstack = (void *)vm_kmalloc(parent->kstacksz);
```

Comment:
`vm_kmalloc` return `void *` type anyway
```suggestion
	current->parentkstack = vm_kmalloc(parent->kstacksz);
```